src/react/controllers/map-controller.js

```diff
@@ -109,18 +113,122 @@ export default class MapController extends PureComponent {
 
   componentWillUpdate(nextProps) {
     this._controls.setOptions(nextProps);
+    this._animateViewportProp(nextProps);
   }
 
   componentWillUnmount() {
     this._eventManager.destroy();
   }
 
-  _onInteractiveStateChange({isDragging = false}) {
+  _onInteractiveStateChange(interactiveState) {
+    const {isDragging = false} = interactiveState;
     if (isDragging !== this.state.isDragging) {
       this.setState({isDragging});
     }
   }
 
+  _extractViewportFromProps(props) {
+    return {
```

Comment:
More of a discussion point: A question is if this component should just take a `viewport` or `viewportState` prop so we don't have to unpack and pack these params again and again. 

This would also allow us to use the various `State` classes directly as input.

We could customize `shouldComponentUpdate` to compare the members of that field.

Reply:
I like this idea, this way it is also easy if a child component contains `viewportState` prop, and only update then.